src/controls/TeamPicker/TeamPicker.tsx

```diff
@@ -0,0 +1,177 @@
+  action: { type: string; payload: any }
+) => {
+  switch (action.type) {
+    case "UPDATE_SELECTEITEM":
+      return { ...state, savedSelectedTeams: action.payload };
+    default:
+      return state;
+  }
+};
+
+// select Team control
+export const TeamPicker : React.FunctionComponent<ITeamPickerProps> = (
+  props: ITeamPickerProps
+) => {
+  // initialize reducer
+  const [state, dispatch] = React.useReducer(reducer, initialState);
+  const picker = React.useRef<IBasePicker<ITag>>(null);
+  const { serviceScope } = props.appcontext;
+  const { getMyTeams } = useTeams(serviceScope);
+  const { pickerStylesMulti, pickerStylesSingle , renderItemStylesMulti, renderItemStylesSingle,renderIconButtonRemoveStyles} = useTeamPickerStyles(theme);
+  const { onSelectedTeams, selectedTeams  , itemLimit, label, styles } = props;
+
+  const useFilterSuggestedTeams = React.useCallback(
+    async (filterText: string, teamsList: ITag[]): Promise<ITag[]> => {
+      let tags: ITag[] = [];
+      try {
+      const teams: ITeam[] = await getMyTeams(filterText);
+
+      if (teams?.length) {
+        for (const team of teams) {
+          const checkExists = find(teamsList,{"key": team.id});
+          if (checkExists) continue;
+          tags.push({ key: team.id, name: team.displayName });
+        }
+      }
+      return tags;
+      } catch (error) {
+        console.log(error);
+        return tags;
+      }
+    },
+    []
+  );
+
+    React.useEffect(()=>{
+      dispatch({
+        type: "UPDATE_SELECTEITEM",
+        payload:  selectedTeams
+      });
+    },[props]);
+
+    const _onRenderItem = React.useCallback((itemProps:IPickerItemProps<ITag>) => {
+      const { savedSelectedTeams  } = state;
+      if (itemProps.item) {
+        return (
+          <Stack
+            horizontal
+            horizontalAlign="start"
+            verticalAlign="center"
+            tokens={{ childrenGap: 7 }}
+            styles={ itemLimit && itemLimit > 1 ?   renderItemStylesMulti : renderItemStylesSingle}
+          >
+            <ImageIcon
+              imageProps={{
+                src:
+                 TEAMS_SVG_LOGO,
+                width: 18,
+                height: 18,
+              }}
+            ></ImageIcon>
+
+            <Text variant="medium">{itemProps.item.name}</Text>
+            <IconButton
+              styles={renderIconButtonRemoveStyles}
+              iconProps={{ iconName: "Cancel" , }}
+              title="remove"
+              onClick={(ev) => {
+                const _newSelectedTeams = pullAllBy(savedSelectedTeams, [
+                  itemProps.item,
+                ]);
+                dispatch({
+                  type: "UPDATE_SELECTEITEM",
+                  payload: _newSelectedTeams,
+                });
+                props.onSelectedTeams(_newSelectedTeams);
+              }}
+            />
+          </Stack>
+        );
+      } else {
+        return null;
+      }
+    },[selectedTeams]);
```

Comment:
I would double-check if we need to include `state` or `state.savedSelectedItems` in debs as well.

Reply:
I include state.savedSelectedItems.